deepchem/models/tensorgraph/layers.py

```diff
@@ -798,8 +846,11 @@ def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
 
 class Input(Layer):
 
-  def __init__(self, shape, dtype=tf.float32, **kwargs):
-    self._shape = tuple(shape)
+  def __init__(self, shape=None, dtype=tf.float32, **kwargs):
+    if shape:
+      self._shape = tuple(shape)
+    else:
+      self._shape = None
```

Comment:
I don't think this is a good idea.  If the shape is unknown, you should leave the field undefined.  `Layer.shape()` checks for that and throws an appropriate exception.  Also, don't you mean `if shape is not None` rather than `if shape`?  The latter is susceptible to lots of bugs if someone passes in a value that just happens to evaluate to false.

Reply:
I guess we need to agree on what the None here means. Does a None mean:

1. We explicitly tell know the shape to be Unknown
or
2. We forgot to set the shape and should throw exceptions?

Reply:
None doesn't mean anything at all.  That isn't part of the API.  Either `_shape` is set to the actual shape, or the field doesn't exist at all.  Setting it to anything other than the actual shape is a bug.